### PR TITLE
Exclude entities

### PR DESCRIPTION
**Pull request by kaareal**
@drakonen 
Change the api excludeEntities used to be an array of objects. Now its an array of strings.

### CODE REVIEW

File: src/models/dimension.js [hunk 1]
```diff
@@ -101,10 +101,13 @@ define([
 
       if (this.get('filter')) {
         var filter = Openhose._.extend(params.filter || {} , this.get('filter'));
-        if (filter.excludeEntities) {
-          filter.excludeEntities = filter.excludeEntities.map(function(item) {
-            return encodeURIComponent(item.entityType) + ':' + encodeURIComponent(item.id);
-          });
+        if (filter.excludeEntities && filter.excludeEntities.length) { 
+          if (typeof filter.excludeEntities[0] !== 'string') {
+            console.warn('DEPRECATED excludeEntities should be an array of strings "<entityType><item.id>" instead of object');
```
Comment (drakonen) [inline src/models/dimension.js]: The colon ":"  is missing in <entityType><item.id>

File: src/models/metric.js [hunk 2]
```diff
@@ -303,10 +303,13 @@ define([
 
       if (this.get('filter')) {
         var filter = Openhose._.extend(params.filter || {} , this.get('filter'));
-        if (filter.excludeEntities) {
-          filter.excludeEntities = filter.excludeEntities.map(function(item) {
-            return encodeURIComponent(item.entityType) + ':' + encodeURIComponent(item.id);
-          });
+        if (filter.excludeEntities && filter.excludeEntities.length) { 
+          if (typeof filter.excludeEntities[0] !== 'string') {
+            console.warn('DEPRECATED excludeEntities should be an array of strings "<entityType><item.id>" instead of object');
```
Comment (drakonen) [inline src/models/metric.js]: The colon ":" is missing in

**Comment by drakonen** [comment]
:+1: :headphones:

File: docs/data.html [hunk 3]
```diff
@@ -843,19 +843,11 @@ <h3 id="filters-exclude-entities">Exclude entities</h3>
     // In this example we wish to know how many messages don't contain #justinbieber,
     // or the topic "Justin Bieber" (refreshing isn't it!)
     // excludeEntities excepts an array because you can exclude multiple
+    // worth noting that the strings need to be escaped like this
+    // encodeURIComponent('topics') + ':' + encodeURIComponent('Justin%20Bieber')
```
Comment (Juice10) [inline docs/data.html]: the space in `encodeURIComponent('Justin%20Bieber')` is already encoded. The description is somewhat confusing, do you mean that everything needs to be double encoded?

File: docs/data.html [hunk 4]
```diff
@@ -843,19 +843,11 @@ <h3 id="filters-exclude-entities">Exclude entities</h3>
...
-        // what is the id of the entity?
-        // In the case of #justinbieber its the following:
-        id: 'justinbieber'
-      },
-      {
-        entityType: 'topics',
-        id: 'Justin Bieber'
-      }
+      'tags:justinbieber'
+      'topics:Justin%20Bieber'
```
Comment (Juice10) [inline docs/data.html]: Please add a description to the items like I did above

File: src/models/dimension.js [hunk 5]
```diff
@@ -101,10 +101,13 @@ define([
...
       if (this.get('filter')) {
         var filter = Openhose._.extend(params.filter || {} , this.get('filter'));
-        if (filter.excludeEntities) {
-          filter.excludeEntities = filter.excludeEntities.map(function(item) {
-            return encodeURIComponent(item.entityType) + ':' + encodeURIComponent(item.id);
-          });
+        if (filter.excludeEntities && filter.excludeEntities.length) { 
+          if (typeof filter.excludeEntities[0] !== 'string') {
+            console.warn('DEPRECATED excludeEntities should be an array of strings "<entityType>:<item.id>" instead of object');
+            filter.excludeEntities = filter.excludeEntities || [].map(function(item) {
```
Comment (Juice10) [inline src/models/dimension.js]: This should be split out so we can use it in multiple places (eg. src/models/metric.js)

**Comment by Juice10** [comment]
Would be good to support both the object notation and the string notation.
Object notation is more descriptive towards users and is loosely coupled from the API, and string notation is easy if you actually know the API intimately

**Pull request closed without merging**